float8_experimental/float8_linear_utils.py

```diff
@@ -66,32 +68,53 @@ def _update_history_with_new_amax(new_amax, amax_history):
 
 
 def swap_linear_with_float8_linear(
-    model,
-    module,
-    emulate=False,
-    skip_fqn_list=None,
-    cur_fqn="",
-):
+    module: nn.Module,
+    module_cls: Type[nn.Module],
+    emulate: bool = False,
+    skip_fqn_list: Optional[List[str]] = None,
+    cur_fqn: str = "",
```

Comment:
@drisspg Do you have historical context on `cur_fqn`?

In particular, did users want to be able to pass `skip_fqn_list` as module FQNs that are "globally" prefixed from the root module, while only call `swap_linear_with_float8_linear` on a submodule?
```
Example:
skip_fqn_list = ["dense_arch.mlp.lin1", "dense_arch.mlp.lin2"]
swap_linear_with_float8_linear(model.dense_arch, ..., skip_fqn_list, cur_fqn="dense_arch")
```

Or, was `cur_fqn` always expected to be `""` upon the user's call to `swap_linear_with_float8_linear`?

If it is the latter case, then personally I would prefer removing this argument as to not leak arguments used for recursion to the user.